### DISCUSSION TITLE
Rendering: objects are removed and added again for each animation frame.

### DISCUSSION THREAD

**Discussion by freddida** [donmccurdy/glTF-Transform#598] [Q&A] [answered]
**Describe the issue**
I did a performance test between native threejs and gtlf-transform-view rendering. I found out that certain operations with view degrade the performance enormously. In my example I scale several cubes in the renderloop.

**Observation**
The objects are removed and added again for each animation frame.

**Question**
Can you explain me the advantage of this method?

If you need further explanation I can provide a minimal example.

**Versions:**
 - Version: [2.1.5]
 - Environment: [Visual Studio Code / Chrome]

**Comment by donmccurdy** [comment, marked as answer]
Hi! While there are conditions that might cause objects to be removed and re-added, changing the scale of a node is not supposed to be one of them. So I think this is probably a bug. Would you mind opening an issue on the https://github.com/donmccurdy/glTF-Transform-View repository with a few more details? I think knowing what code you are running in the render loop (just `node.setScale([x, y, z])`, or more?) might be enough. Thanks!

> EDIT: tracking in https://github.com/donmccurdy/glTF-Transform-View/issues/19
